flaskr/backservice.py

```diff
@@ -0,0 +1,181 @@
+import os
+import psycopg2
+from dotenv import load_dotenv
+from flask import Flask, request, make_response, render_template, jsonify
+from flaskr import helper
+
+load_dotenv()
+
+
+# region DB CONNECTION VARIABLES
+DB_HOST = os.getenv("DB_HOST") 
+DB_PORT = os.getenv("DB_PORT")
+DB_NAME = os.getenv("DB_NAME")
+DB_USERNAME = os.getenv("DB_USERNAME")
+DB_PASSWORD = os.getenv("DB_PASSWORD")
+dbConnectionOptions = "-c statement_timeout={}".format(os.getenv("DB_TIMEOUT"))
+connection = object
+# endregion
+# region SQL QUERIES PARAMS
+RECORDS_LIMIT = os.getenv("RECORDS_LIMIT")
+# The API request will work on this table with this primary key column
+TABLE_NAME = os.getenv("TABLE_NAME")
+# The private key column name of the TABLE_NAME
+KEY_COLUMNS_NAMES = os.getenv("KEY_COLUMNS_NAMES")
+# endregion
+DB_NOT_WORKING_MSG = f"Connection failed with HOST={DB_HOST}, PORT={DB_PORT}, NAME={DB_NAME}, USERNAME={DB_USERNAME}, password is stored in secret."
+CONTENT_HEADER = {"content_type": "application/json; charset=utf-8"}
+
+app = Flask(__name__)
+
+
+@app.errorhandler(404) 
+def invalid_route(e): 
+    return make_response(jsonify({'msg' : 'Route not found'}), 404, CONTENT_HEADER)
+
+@app.errorhandler(503)
+def server_unavailable(e):
+    return make_response(jsonify({'error': 'Server is unavailable'}), 503, CONTENT_HEADER)
+
+
+@app.route("/")
+def homePage():
+    # TODO: MAKE SOMETHING ELSE WITH THIS
+    return render_template("home.html")
+
+
+@app.route("/health", methods=["GET"])
+def apiHandelHealth():
+    return make_response(jsonify({"msg": "Healthy"}), 200, CONTENT_HEADER)
+
+
+@app.route("/ready", methods=["GET"])
+def apiHandelReady():
+    connection = None
+    try:
+        connection = psycopg2.connect(host=DB_HOST, port=DB_PORT, dbname=DB_NAME, user=DB_USERNAME, password=DB_PASSWORD, options = dbConnectionOptions) 
+        if (connection.status):
+            return make_response(jsonify({"msg": "DB connection work"}), 200, CONTENT_HEADER)  
+        else: 
+            return make_response(jsonify({"msg": f"DB connection does not work: \n {DB_NOT_WORKING_MSG}"}), 404, CONTENT_HEADER)
+    except (psycopg2.OperationalError) as error:
+        return make_response(jsonify({"msg": f"{DB_NOT_WORKING_MSG} \n One or more of the connection params is incorrect. \n {error}"}), 400, CONTENT_HEADER)
+    except (psycopg2.Error) as error:             
+        return make_response(jsonify({"msg": f"{DB_NOT_WORKING_MSG} \n {error}"}), 400, CONTENT_HEADER)
+        #TODO: LOG THE ERROR
+    finally:
+        if(connection is not None):
+            connection.close()
+
+
+# Return all the records that match the args parameters
+def createQueryByRequestType(args: dict, request_type: str, action: str = None):
```

Comment:
I agree, it can be a little complex. 
I was trying to prevent a duplication of code and makeing it generic by doing the connectivity check, the connection and the exceptions handeling only once

Reply:
It is a question of duplication over readability